### PR TITLE
Add 5 new ABI breaking test cases (69–73)

### PR DESCRIPTION
**Pull request by napetrov**
## Summary

Add five new ABI breaking test cases covering calling convention changes, flexible array members, inline namespace versioning, class hierarchy insertion, and typedef widening.

## Motivation

Expand the ABI compliance test suite with additional edge cases that represent real-world breaking changes:
- **case69**: Trivially-copyable to non-trivial struct (calling convention change)
- **case70**: Flexible array member element type change
- **case71**: Inline namespace version bump (symbol mangling change)
- **case72**: Class hierarchy insertion (vtable and covariant return changes)
- **case73**: Typedef underlying type widening

These cases test detection of subtle but critical ABI breaks that are often missed by simple size/offset comparisons.

## Changes

- Added 5 new test case directories with v1/v2 implementations:
  - `case69_trivial_to_nontrivial/`: Struct gains destructor → register passing becomes hidden pointer
  - `case70_flexible_array_member_changed/`: FAM element type float → double
  - `case71_inline_namespace_moved/`: Functions moved between inline namespaces → symbol name change
  - `case72_covariant_return_changed/`: Intermediate class inserted → vtable shift and covariant return change
  - `case73_typedef_underlying_changed/`: Typedef int → void* → size and calling convention change

- Each case includes:
  - v1.h/v1.cpp (or .c): Original implementation
  - v2.h/v2.cpp (or .c): Breaking change
  - app.cpp (or .c): Test application
  - README.md: Detailed explanation of the break, real failure demo, and how to fix
  - CMakeLists.txt: Build configuration

- Updated `ground_truth.json` with expected verdicts and detection kinds for all 5 cases

- Updated main `README.md` to reflect 74 total published cases

## Checklist

- [x] Tests added (5 new complete test cases with v1/v2/app implementations)
- [x] Docs added (comprehensive README.md for each case with real failure demos)
- [x] `ground_truth.json` updated with expected verdicts and detection kinds
- [x] All cases follow existing case structure and naming conventions

https://claude.ai/code/session_01KH3yANjMUnijThFWQWrMda

## Summary by CodeRabbit

* **New Features**
  * Added five new ABI test cases (69–73) covering breaking scenarios: value-ABI trait change, flexible-array-member element change, inline-namespace move, covariant-return/class-hierarchy change, and typedef underlying-type change.

* **Documentation**
  * Added detailed per-case READMEs with explanations, examples, and real-failure demos for each new case.

* **Tests**
  * Updated ground-truth and unit test expectations to reflect 74 published cases.

### CODE REVIEW

File: examples/ground_truth.json [hunk 1]
```diff
@@ -844,6 +844,76 @@
       "bad_practice": false,
       "description": "Virtual method added to previously non-virtual class \u2014 vtable pointer inserted, sizeof grows, all field offsets shift (FUNC_VIRTUAL_ADDED)",
       "expected_kinds": ["func_virtual_added"]
+    },
+    "case69_trivial_to_nontrivial": {
+      "expected": "BREAKING",
```
[the remark this answers is not in the document]
Comment (napetrov) [reply examples/ground_truth.json:849]: Fixed in 43d4ef4 — updated `_EXPECTED_CASE_COUNT` from 69 to 74.

File: examples/ground_truth.json [hunk 2]
```diff
@@ -844,6 +844,76 @@
...
+      "platforms": [
+        "linux",
+        "macos",
+        "windows"
+      ],
+      "abi_break": true,
+      "api_break": true,
+      "bad_practice": false,
+      "description": "Flexible array member element type changed float \u2192 double \u2014 dynamic allocation size and element access offsets break while sizeof(Packet) stays the same (FLEXIBLE_ARRAY_MEMBER_CHANGED)",
+      "expected_kinds": ["flexible_array_member_changed"]
```
[the remark this answers is not in the document]
Comment (napetrov) [reply examples/ground_truth.json]: Fixed in 43d4ef4 — added `func_return_changed` (which is BREAKING in the registry) to case70's `expected_kinds`. The `packet_sum` return type changes from `float` to `double`, so this kind is legitimately detected alongside `flexible_array_member_changed`.
